pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -902,6 +903,17 @@ func defaultGCP(m *Machine, config *admissionConfig) (bool, []string, utilerrors
 
 	providerSpec.Disks = defaultGCPDisks(providerSpec.Disks, config.clusterID)
 
+	if len(providerSpec.GuestAccelerators) != 0 {
+		// In case AcceleratorCount was not set it should default to 1, since there is no valid reason for it to be purposely set to 0.
+		if providerSpec.GuestAccelerators[0].AcceleratorCount == 0 {
+			providerSpec.GuestAccelerators[0].AcceleratorCount = defaultGCPAcceleratorCount
+		}
+	}
+
+	if len(providerSpec.GuestAccelerators) != 0 || strings.HasPrefix(providerSpec.MachineType, "a2-") {
+		providerSpec.OnHostMaintenance = "TERMINATE"
+	}
```

Comment:
Does this make sense? Are we sure we want to override this on behalf of a user? I'd prefer we set an error/warning and then users can update it themselves right?

If this isn't set, does the instance definitely fail to create?